api/tests/unit/domain/usecases/get-campaign-participation-result_test.js

```diff
@@ -38,62 +43,229 @@ describe('Unit | UseCase | get-campaign-participation-result', () => {
+      beforeEach(() => {
+        // given
+        badgeRepository.findOneByTargetProfileId.withArgs(targetProfileId).resolves(badge);
+        badgeAcquisitionService.reviewBadgeCriterias.withArgs({ campaignParticipationResult }).resolves(true);
       });
 
-      // then
-      expect(actualCampaignParticipationResult).to.deep.equal(campaignParticipationResult);
+      it('should create the badge acquisition', async () => {
+        // when
+        await getCampaignParticipationResult({
+          userId,
+          campaignParticipationId,
+          campaignParticipationRepository,
+          campaignRepository,
+          targetProfileRepository,
+          competenceRepository,
+          assessmentRepository,
+          badgeRepository,
+          knowledgeElementRepository,
+          badgeAcquisitionRepository,
+          badgeAcquisitionService,
+        });
+
+        // then
+        sinon.assert.calledOnce(badgeAcquisitionRepository.create);
+        sinon.assert.calledWith(badgeAcquisitionRepository.create, {
+          badgeId: campaignParticipationResult.badge.id ,
+          userId
+        });
+      });
+
+      it('should get the campaignParticipationResult', async () => {
+        // when
+        const actualCampaignParticipationResult = await getCampaignParticipationResult({
+          userId,
+          campaignParticipationId,
+          campaignParticipationRepository,
+          campaignRepository,
+          targetProfileRepository,
+          competenceRepository,
+          assessmentRepository,
+          badgeRepository,
+          knowledgeElementRepository,
+          badgeAcquisitionRepository,
+          badgeAcquisitionService,
+        });
+
+        // then
+        expect(actualCampaignParticipationResult).to.deep.equal(campaignParticipationResult);
+      });
+    });
+
+    context('when a badge is not available for the campaignParticipationResult', () => {
+      beforeEach(() => {
+        // given
+        badgeRepository.findOneByTargetProfileId.withArgs(targetProfileId).resolves({});
+        badgeAcquisitionService.reviewBadgeCriterias.withArgs({ campaignParticipationResult }).resolves(false);
+      });
+
+      it('should not create the badge acquisition', async () => {
+        // when
+        await getCampaignParticipationResult({
+          userId,
+          campaignParticipationId,
+          campaignParticipationRepository,
+          campaignRepository,
+          targetProfileRepository,
+          competenceRepository,
+          assessmentRepository,
+          badgeRepository,
+          knowledgeElementRepository,
+          badgeAcquisitionRepository,
+          badgeAcquisitionService,
+        });
+
+        // then
+        sinon.assert.notCalled(badgeAcquisitionRepository.create);
+      });
+
+      it('should get the campaignParticipationResult', async () => {
+        // when
+        const actualCampaignParticipationResult = await getCampaignParticipationResult({
```

Comment:
J'ai du mal à lire les tests avec ce bloc d'appel dans la partie `when`. Que penses-tu de déplacer ce code dans une fonction ? Il semblerait que les arguments passés soient toujours les mêmes. Cela réduirait drastiquement le nombre de lignes dans ce fichier.

Reply:
Si je comprend bien, tu aimerais quelque chose comme ça ? 
```suggestion
		// en global
		const usecaseDependancies = {
          userId,
          campaignParticipationId,
          campaignParticipationRepository,
          campaignRepository,
          targetProfileRepository,
          competenceRepository,
          assessmentRepository,
          badgeRepository,
          knowledgeElementRepository,
          badgeAcquisitionRepository,
          badgeAcquisitionService,
        };
        
        // dans chaque test
        const actualCampaignParticipationResult = await getCampaignParticipationResult(usecaseCependancies);
```

Reply:
Oui exactement 👍